artifactory.py

```diff
@@ -377,17 +377,30 @@ def __exit__(self, *excinfo):
 
 
 def quote_url(url):
+    """
+    Quote URL to allow URL fragment identifier as artifact folder or file names.
+    See https://en.wikipedia.org/wiki/Percent-encoding#Reserved_characters
+    Function will percent-encode the URL
+
+    Note: if anybody for some reason (actually this should not be allowed) will require = or ; sign in artifact name,
+    then have to change this function
+    :param url: (str) URL that should be quoted
+    :return: (str) quoted URL
+    """
     parsed_url = urllib3.util.parse_url(url)
     if parsed_url.port:
         quoted_path = requests.utils.quote(
-            url.rpartition(f"{parsed_url.host}:{parsed_url.port}")[2]
+            url.rpartition(f"{parsed_url.host}:{parsed_url.port}")[2],
+            safe="/;=",  # ; and = signs are used in Artifactory Matrix Parameters
```

Comment:
Hi @beliaev-maksim 

If we ignore them, there would be issues if artifacts contains ';' or '='.

Similar to [Jfrog Cli](https://www.jfrog.com/confluence/display/CLI/CLI+for+JFrog+Artifactory#CLIforJFrogArtifactory-UploadingFiles), how about making property as an initilization parameter and appending property at the end of URL before accessing Artifactory?
e.g.
```Python
path = ArtifactoryPath("http://my-artifactory/artifactory/libs-snapshot-local/myapp/1.0",
    props={'foo1': 'bar1', 'foo2': 'bar2'})
```

Mixing the Artifactory Path and properties may not be a better solution.

Reply:
That is what I wrote in the comment to the function

However, I don't expect = or ; sign in the name, that is too unconventional

Reply:
But actually, there is artifact name contains this kind of special characters, such as '-', ':' in closed issues.
I don't think they are conventional either, but we cannot make this decision for users. Besides, Artifactory support special characters. 

I mean could we append properties after `quote_url` instead of before `quote_url`?
https://github.com/devopshq/artifactory/blob/3149895191e3a03d727f752c49428f05931b7fbe/artifactory.py#L1018

Reply:
I will have a look how to handle it differently

anyway I need #272 to be merged first

Reply:
I just want to let you know my concern.

Go ahead to merge it.

Reply:
matrix params added in request after encoding